### PR TITLE
DP-26237 [SI-a11y] Collapsible content label

### PR DESCRIPTION
**Pull request by ygannett**
Any PRs being created needs a changelog.txt file before being merged into dev. See: [Change Log Instructions](https://github.com/massgov/mayflower/blob/develop/docs/for-developers/changelog-instructions.md)

## Description

- Correct the markup, it's invalid that a heading is nested to a button.
- Clean up confusing aria-label content
- Add context for the item count to make sense to screen reader users
- Hide "+/-" accordion state icon from screen readers

## Related Issue / Ticket

- [JIRA issue](https://massgov.atlassian.net/browse/DP-26237)
- [Github issue]()

## Steps to Test

1. Go to /patternlab/patterns/03-organisms-by-author-collapsible-content-extended/03-organisms-by-author-collapsible-content-extended.html
2. In inspect, find:
    - `button` is in a heading.
    - "+" symbol on the right in the button is not css content.
    - No `aria-label` on `button`s.
3. Check `aria-expanded` gets updated as open/close the accordions.
4. No visual change.

## Screenshots
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.

## Additional Notes:

Anything else to add?

#### Impacted Areas in Application

*

#### @TODO

*

### CODE REVIEW

File: packages/patternlab/styleguide/source/_patterns/03-organisms/by-author/collapsible-content.json [hunk 1]
```diff
@@ -23,6 +23,7 @@
       "collapsibleHeader": {
         "info": "Click here to expand this thing",
         "title": "Collapsible Header",
+        "state": "ttue",
```
Comment (clairesunstudio) [inline packages/patternlab/styleguide/source/_patterns/03-organisms/by-author/collapsible-content.json]:
```suggestion
        "state": true,

```

File: packages/patternlab/styleguide/source/_patterns/02-molecules/collapsible-header.json [hunk 2]
```diff
@@ -2,6 +2,7 @@
   "collapsibleHeader": {
     "info": "Click here to read more about this thing",
     "title": "Collapsible Header",
+    "state": "false",
```
Comment (clairesunstudio) [inline packages/patternlab/styleguide/source/_patterns/02-molecules/collapsible-header.json]: can you add this new prop in the component markdown doc?

Comment (clairesunstudio) [reply packages/patternlab/styleguide/source/_patterns/02-molecules/collapsible-header.json]: I think we should make it a boolean, to be consistent. What do you think?

File: changelogs/DP-26237.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+Changed:
+  - project: Patternlab
+    component: collapsibleHeader
```
Comment (clairesunstudio) [inline changelogs/DP-26237.yml]:
```suggestion
    component: CollapsibleHeader
```

File: changelogs/DP-26237.yml [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+    description: Correct invalid markup, cleaned up confusing aria-label, add context for screenr reader users. (#1703)
```
Comment (clairesunstudio) [inline changelogs/DP-26237.yml]:
```suggestion
    description: Correct invalid markup, cleaned up confusing aria-label, add context for screen reader users.  Add a new prop `state` for specifying the accordion's default state. (#1703)
```

**Comment by ygannett** [comment]
Backstop failures are not related to this PR's changes.

File: packages/patternlab/styleguide/source/_patterns/02-molecules/collapsible-header.json (hunk 2 above)
Comment (ygannett) [reply packages/patternlab/styleguide/source/_patterns/02-molecules/collapsible-header.json]: Yes, that makes sense.  The state is added as boolean to the MD.  Thank you for mentioning that.

**Review by clairesunstudio — APPROVED** [review]

**Pull request merged**